packages/next/src/build/swc/index.ts

```diff
@@ -623,38 +623,47 @@ function bindingToApi(
       ...options,
       nextConfig: await serializeNextConfig(
         options.nextConfig,
-        path.join(options.rootPath, options.projectPath)
+        path.join(options.rootPath, options.projectPath),
+        options.dev
       ),
       env: rustifyEnv(options.env),
     }
   }
 
   async function rustifyPartialProjectOptions(
-    options: Partial<ProjectOptions>
+    options: Partial<ProjectOptions>,
+    dev: boolean
   ): Promise<NapiPartialProjectOptions> {
     return {
       ...options,
       nextConfig:
         options.nextConfig &&
         (await serializeNextConfig(
           options.nextConfig,
-          path.join(options.rootPath!, options.projectPath!)
+          path.join(options.rootPath!, options.projectPath!),
+          dev
         )),
       env: options.env && rustifyEnv(options.env),
     }
   }
 
   class ProjectImpl implements Project {
     private readonly _nativeProject: { __napiType: 'Project' }
+    private dev: boolean
```

Comment:
I need to get the current value from somewhere so that the `update` implementation can handle calls where `dev` isn't in the new options.

Not sure if we can make that readonly and then just throw if you try to switch between dev and prod in `update` or if switching between dev and prod is legit in `update`